src/api/app/controllers/webui/package_controller.rb

```diff
@@ -980,8 +975,9 @@ def rpmlint_result
         end
       end
     end
-    repos.uniq.each do |repo_name|
-      @repo_list << [repo_name, valid_xml_id(elide(repo_name, 30))]
+
+    @repo_list = repos.uniq.collect do |repo_name|
```

Comment:
[nitpick]  using `collect!` save some memory :wink: and now that you are refactoring this you could go a bit further and get rid of repos, call it `@repo_list` instead and directly do `@repo_list.uniq!.collect!` which saves even more memory. :wink: :wink:

Reply:
ah actually, I always pursue immutability and Nil avoidance, and therefore I prefer it like that than this "micro optimization" (for memory, yes, speed, no)

Reply:
well memory allocation needs time... but it is a nitpick